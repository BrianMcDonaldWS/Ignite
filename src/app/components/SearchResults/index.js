import React from 'react';
import PropTypes from 'prop-types';
import makeClass from 'classnames';

import SearchResult from '../SearchResult';
import styles from './search-results.css';

function SearchResults({ searchResults }) {
  return (
    <div className={makeClass(styles.searchResults)}>
      {searchResults.map(([fileName, results]) => (
        <SearchResult
          key={fileName}
          setResults={searchResults => this.setState({ searchResults })}
          fileName={fileName}
          results={results}
        />
      ))}
    </div>
  );
}

SearchResults.propTypes = {
  searchResults: PropTypes.array
};

SearchResults.defaultProps = {
  searchResults: []
};

export default SearchResults;
