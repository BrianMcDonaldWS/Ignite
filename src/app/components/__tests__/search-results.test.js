import React from 'react';

import SearchResults from '../SearchResults';
import renderToJson from './utils/render-to-json';

test('Icon', () => {
  expect(
    renderToJson(
      <SearchResults
        searchResults={[
          ['foo.md', ['text results for foo.md']],
          ['bar.md', ['text results for bar.md']]
        ]}
      />
    )
  ).toMatchSnapshot();
});
