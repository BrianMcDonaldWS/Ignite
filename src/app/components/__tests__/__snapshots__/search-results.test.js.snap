// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 1`] = `
<div
  className="searchResults"
>
  <SearchResult
    fileName="foo.md"
    results={
      Array [
        "text results for foo.md",
      ]
    }
    setResults={[Function]}
  />
  <SearchResult
    fileName="bar.md"
    results={
      Array [
        "text results for bar.md",
      ]
    }
    setResults={[Function]}
  />
</div>
`;
